process/accounts/serialize.go

```diff
@@ -89,26 +90,16 @@ func prepareSerializedAccountInfo(
 	return meta, serializedData, nil
 }
 
-func encodeNonceToHex(nonce uint64) string {
-	if nonce == 0 {
-		return "00"
-	}
-
-	nonceBigBytes := big.NewInt(0).SetUint64(nonce).Bytes()
-	hexEncodedNonce := hex.EncodeToString(nonceBigBytes)
-
-	return hexEncodedNonce
-}
-
 // SerializeAccountsHistory will serialize accounts history in a way that Elastic Search expects a bulk request
 func (ap *accountsProcessor) SerializeAccountsHistory(
```

Comment:
Instead of adding a bool var (which signal that the function is doing more than one thing), I would completely create a new function called `SerializeAccountsHistoryForESDT` and reduce the duplicated code

Reply:
refactored this.